# Análise Inicial dos Dados

In [6]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

print("Bibliotecas carregadas com sucesso!")

Bibliotecas carregadas com sucesso!


In [16]:
df = pd.read_csv('healthcare_dataset.csv') # Lendo o arquivo e criando o DataFrame

df.head() # Exibindo as primeiras linhas do DataFrame para verificar os dados

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results
0,Bobby JacksOn,30,Male,B-,Cancer,2024-01-31,Matthew Smith,Sons and Miller,Blue Cross,18856.281306,328,Urgent,2024-02-02,Paracetamol,Normal
1,LesLie TErRy,62,Male,A+,Obesity,2019-08-20,Samantha Davies,Kim Inc,Medicare,33643.327287,265,Emergency,2019-08-26,Ibuprofen,Inconclusive
2,DaNnY sMitH,76,Female,A-,Obesity,2022-09-22,Tiffany Mitchell,Cook PLC,Aetna,27955.096079,205,Emergency,2022-10-07,Aspirin,Normal
3,andrEw waTtS,28,Female,O+,Diabetes,2020-11-18,Kevin Wells,"Hernandez Rogers and Vang,",Medicare,37909.782410,450,Elective,2020-12-18,Ibuprofen,Abnormal
4,adrIENNE bEll,43,Female,AB+,Cancer,2022-09-19,Kathleen Hanna,White-White,Aetna,14238.317814,458,Urgent,2022-10-09,Penicillin,Abnormal


In [17]:
print("--- Resumo de Tipos e Nulos ---") 
df.info() # Resumo dos tipos de dados e valores nulos

print("--- Contagem de Valores Nulos ---")
print(df.isnull().sum()) # Contagem de valores nulos por coluna

--- Resumo de Tipos e Nulos ---
<class 'pandas.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Name                55500 non-null  str    
 1   Age                 55500 non-null  int64  
 2   Gender              55500 non-null  str    
 3   Blood Type          55500 non-null  str    
 4   Medical Condition   55500 non-null  str    
 5   Date of Admission   55500 non-null  str    
 6   Doctor              55500 non-null  str    
 7   Hospital            55500 non-null  str    
 8   Insurance Provider  55500 non-null  str    
 9   Billing Amount      55500 non-null  float64
 10  Room Number         55500 non-null  int64  
 11  Admission Type      55500 non-null  str    
 12  Discharge Date      55500 non-null  str    
 13  Medication          55500 non-null  str    
 14  Test Results        55500 non-null  str    
dtypes: float64(1), int64(2), str(12)

In [ ]:
print(df.to_string()) # Exibindo o DataFrame completo para análise detalhada

 **Analisando este dataframe, pude perceber alguns pontos a trabalhar para melhorar a qualidade dos dados:**
 
 Na coluna 'Name' há muitos nomes que variam entre letra minúscula e maiúscula

 Outra coisa que identifiquei foi que as colunas de 'Date' estão em formato de string

In [22]:
df.describe()

,Age,Date of Admission,Billing Amount,Room Number,Discharge Date
count,55500.000000,55500,55500.000000,55500.000000,55500
mean,51.539459,2021-11-01 01:02:22.443243,25539.316097,301.134829,2021-11-16 13:15:20.821621
min,13.000000,2019-05-08 00:00:00,-2008.492140,101.000000,2019-05-09 00:00:00
25%,35.000000,2020-07-28 00:00:00,13241.224652,202.000000,2020-08-12 00:00:00
50%,52.000000,2021-11-01 00:00:00,25538.069376,302.000000,2021-11-17 00:00:00
75%,68.000000,2023-02-03 00:00:00,37820.508436,401.000000,2023-02-18 00:00:00
max,89.000000,2024-05-07 00:00:00,52764.276736,500.000000,2024-06-06 00:00:00
std,19.602454,NaN,14211.454431,115.243069,NaN


**Exercício 1: Análise de Age**
1. A idade mínima é 13 e a máxima é 89.
   - Qual é a "largura" (diferença) entre min e max?
   
2. Compare:
   - Mean = ?
   - 50% (mediana) = ?
   - Eles são parecidos ou muito diferentes?
   
3. Olhe para o std:
   - É alto ou baixo comparado com a média?
   - (Dica: divide std pela mean e pense se é grande ou pequeno)

---------------------------------------------------------------------------------------------------------------------

Respostas:

1. A diferença entre ambas é de 76 anos de diferença, o que significa que trabalhamos com dados de muitas faixas etárias diferentes aparentemente.

2. Média = 51.3 e Mediana = 52, isto significa que temos dados equilibrados, ou seja, não temos idades muito avançadas ou novas pesando nossa média/mediana.

_3. Acredito que é baixo comparado com a média, isso deve significar que os dados tem dados consistentes._ --- 
Correção: Na realidade o número é alto, isso nos mostra que os dados estão moderamente espalhados.
 

**Exercício 2: Billing Amount**

1. Valor mínimo: -2008.49
   - Faz sentido? (Dica: pode ser reembolso ou erro?)

2. Compare Mean vs Mediana:
   - Mean = 25539.32
   - Mediana (50%) = 25538.07
   - O que significa estarem tão próximos?

3. Std = 14211.45
   - Razão std/mean = 14211 ÷ 25539 ≈ ?
   - É alto ou baixo? O que significa para os valores de cobrança?

--------------------------------------------------------------------------------------------------------------------

Respostas:

1. A princípio pensei que poderia ser algum erro no banco de dados, porém com a dica que me foi passada, creio que faz sentido poder ser reembolso. Mas para poder ter certeza disto, é necessário realizar um filtro de quantos valores negativos temos, para analisar se é algo que se repete, tendo mais chances de realmente ser um reembolso.

2. A média e mediana se encontram bem próximas, isso significa que os dados estão consistentes, e não temos nenhum valores significantes interferindo para cima ou para baixo.

3. Desvio Padrão/Média = 55%, isto é muito alto significando que os valores estão espalhados.

In [24]:
# Investigar valores negativos no Billing Amount
valores_negativos = df[df['Billing Amount'] < 0]
print(f"Quantidade de valores negativos: {len(valores_negativos)}")
print(f"Porcentagem do total: {len(valores_negativos)/len(df)*100:.2f}%")
print("\nPrimeiros 5 valores negativos:")
print(valores_negativos[['Name', 'Billing Amount', 'Medical Condition']].head())

Quantidade de valores negativos: 108
Porcentagem do total: 0.19%

Primeiros 5 valores negativos:
                   Name  Billing Amount Medical Condition
132     Ashley Erickson     -502.507813            Cancer
799   Christopher Weiss    -1018.245371            Asthma
1018      Ashley Warner     -306.364925      Hypertension
1421       Jay Galloway     -109.097122            Asthma
2103  Joshua Williamson     -576.727907          Diabetes


Com isso podemos ter a confirmação de que com toda esta quantidade de valores negativos, a maior possibilidade é deles serem reembolsos ao invés de erros.

**Exercício 3: Room Number**

1. Valores: min=101, mean=301.13, max=500
   - Faz sentido para numeração de quartos de hospital?
   - Que tipo de padrão você observa nos números?

2. Std = 115.24
   - Razão std/mean = 115.24 ÷ 301.13 ≈ ?
   - Comparado com Age (38%) e Billing Amount (56%), como está?

3. Distribuição geral:
   - Olhando min=101, max=500: quantos quartos diferentes existem?
   - A média estar em 301 faz sentido?

-------------------------------------------------------------------

Respostas:

1. Faz sentido considerarmos que o primeiro quarto é o 101 e último é 500, com isso podemos ter a percepção que a média em 300 fica no meio do valores de ambos os quartos. Isto significa que temos registros consistentes, não só por este fato que exemplifiquei, mas também pela mediana ser 302.

2. Nosso std é moderadamente alto, 38,3%, comparando com 'Age' e 'Billing Amount', podemos dizer que mantém um padrão dos stds dos dados até o momento.

3. Para ter um dado preciso de quantos quartos diferentes existem, seria mais fácil realizar a contagem distinta para cada um. Não posso fornecer algo preciso, analisando pelo mínimo e máximo descrito. Porém a média estar em 301 faz sentido, baseado nos dados analisados até o momento.

In [26]:
# Verificar quartos únicos
print(f"Total de quartos possíveis (min-max): {df['Room Number'].max() - df['Room Number'].min() + 1}")
print(f"Total de quartos únicos usados: {df['Room Number'].nunique()}")
print(f"Porcentagem de ocupação: {df['Room Number'].nunique() / (df['Room Number'].max() - df['Room Number'].min() + 1) * 100:.1f}%")

Total de quartos possíveis (min-max): 400
Total de quartos únicos usados: 400
Porcentagem de ocupação: 100.0%


Com este código, podemos confirmar a quantidade de quartos.